src/services/cart.js

```diff
@@ -0,0 +1,111 @@
+const  { Cart, CartItem  }= require('../config');
+
+class CartService{
+    async createCart (userId){
+        //Create the cart adding the User id as FK
+        console.log('Id', userId);
+        const cart =Cart.build({ 
+            status: 'new',
+            total: 0,
+            date: new Date(),
+            UserId: userId
+        });
+        await cart.save();
+        return cart
+    }
+    
+    async findAllCarts(){
+        const carts = await Cart.findAll();
+        if (carts===null){
+            throw new Error('There are no existing Carts');
+        }else{
+            return carts;
+        }
+    }
+    
+    async findCartbyId(id){
+        //find the Cart by his own Id
+        const cart = await Cart.findByPk(id,{ include: [CartItem] });
+        if (cart===null){
+            throw new Error('Cart not found');
+        }else{
+            return cart;
+        }
+    }
+
+    async findCartbyUserId(userData) {
+        //Find the Cart by the User Id that owns the cart. 
+        const cart = await Cart.findOne({where: { UserId: userData.UserId}});  
+        if (cart===null){
+            throw new Error('User cart not found');
+        }else{
+            const data = await this.findCartbyId(cart.id);
+            return data;
+        }
+    }
+    
+    
+    async updateCart(data){
+        //update the Cart Data based on the values given
+        await Cart.update(
+            {
+                status: data.status,
+                total: data.total,
+                date: new Date()
+            },
+            {
+                where: {id:data.id}
+            }
+        );
+        const cart = await Cart.findAll({where:{id:data.id}})
+        if (cart===null){
+            throw new Error('Cart not found');
+        }else{
+            return cart;
+        }
```

Comment:
this can be replace with 

```
const cart = await Cart.update(
            {
                status: data.status,
                total: data.total,
                date: new Date()
            },
            {
                where: {id:data.id}
            }
        );
```

Reply:
same problem as the one above, aside of that, i can remove the if and else part if you find them to not be necessary (same for the If/else from lines 98-102, the ones from the review above).